code/modules/research/xenobiology/xenobio_camera.dm

```diff
@@ -138,6 +142,33 @@
 	UnregisterSignal(user, COMSIG_XENO_MONKEY_CLICK_CTRL)
 	..()
 
+/obj/machinery/computer/camera_advanced/xenobio/proc/insert_potion(obj/item/slimepotion/slime/potion)
+	clear_potion()
+	potion.forceMove(src)
+	current_potion = potion
+	RegisterSignal(current_potion, COMSIG_PARENT_QDELETING, PROC_REF(clear_potion))
+
+/obj/machinery/computer/camera_advanced/xenobio/proc/clear_potion()
+	if(!QDELETED(current_potion))
+		current_potion.forceMove(drop_location())
+		UnregisterSignal(current_potion, COMSIG_PARENT_QDELETING)
+	current_potion = null
+
+/obj/machinery/computer/camera_advanced/xenobio/proc/capture_slime(mob/living/simple_animal/slime/slime)
+	slime.visible_message("<span class='notice'>[slime] vanishes in a flash of light!</span>")
+	slime.forceMove(src)
+	stored_slimes += slime
+	RegisterSignal(slime, COMSIG_PARENT_QDELETING, PROC_REF(clear_slime))
+
+/obj/machinery/computer/camera_advanced/xenobio/proc/release_slime(mob/living/simple_animal/slime/slime, release_spot)
+	slime.visible_message("<span class='notice'>[slime] warps in!</span>")
+	slime.forceMove(release_spot)
+	clear_slime(slime)
```

Comment:
```suggestion
	clear_slime(slime)
	slime.forceMove(release_spot)
```
For the very off chance that the forceMove results in a delete.

Reply:
fixed in next commit